frontend/src/main/java/com/google/tripmeout/frontend/servlets/PlaceIndividualServlet.java

```diff
@@ -0,0 +1,154 @@
+package com.google.tripmeout.frontend.servlets;
+
+import com.google.gson.Gson;
+import com.google.maps.GeoApiContext;
+import com.google.maps.PlaceDetailsRequest;
+import com.google.maps.errors.InvalidRequestException;
+import com.google.maps.model.PlaceDetails;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.error.PlaceVisitNotFoundException;
+import com.google.tripmeout.frontend.error.TripMeOutException;
+import com.google.tripmeout.frontend.storage.PlaceVisitStorage;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class PlaceIndividualServlet extends HttpServlet {
+  private final PlaceVisitStorage placeStorage;
+  private final Gson gson;
+
+  private static final Pattern URI_NAME_PATTERN =
+      Pattern.compile(".*/trips/([^/]+)/placeVisits/([^/]+)");
+
+  private final GeoApiContext CONTEXT =
```

Comment:
Also, the API key itself should probably be set from an environment variable (or otherwise not-checked-in config property) by the module that wires in the new Maps service(s). We can put that service (and its module and associated data types) into a new package.

Reply:
also also also, this key won't do what you want. This is the static embeded maps API key. You need one that works for t the Places API I think...